src/account/__tests__/accountsReducer-test.js

```diff
@@ -13,10 +13,11 @@ import accountsReducer from '../accountsReducer';
 import { ZulipVersion } from '../../utils/zulipVersion';
 
 import * as eg from '../../__tests__/lib/exampleData';
+import { makeUserId } from '../../api/idTypes';
 
 describe('accountsReducer', () => {
   describe('REALM_INIT', () => {
-    const account1 = eg.makeAccount();
+    const account1 = eg.makeAccount({ userId: makeUserId(4) });
```

Comment:
Interesting; why do we need to set `account1`'s user ID to 4?

Reply:
Earlier the account Id was unaffected in the redux state, in this change the `REALM_INIT` action modifies it to whatever user ID is provided in the payload. In these test cases the example data provides a user ID of 4 (line 657 of `exampleData.js`) hence the need to keep it 4 while initializing the account. 

In contrast, earlier we used to generate and expect a random user ID but that would always be changed to 4 in this case. 4 could be any other number as long as it is the same in this file and in `exampleData.js`.

Reply:
Hmm, does this test actually depend on using the same ID as in that line of `exampleData`?

If so, we should fix that. In general, we want the assumptions that a test is making to be clear when just reading the test. If it depends on any facts about test data elsewhere, like in `exampleData`, then it should only depend on that data being "reasonable" and boring, not on any arbitrary details.

For example we should probably make `eg.action.realm_init.user_id` be `eg.selfUser.userId`. If we did, it'd be fine for tests to assume that was the case, because in general we make a lot of the `exampleData` data consistent with each other, and that's the only particular value it could have that would be boring and reasonable for it to have. But otherwise, tests shouldn't assume any particular value there at all.

Reply:
Yeah, I see, it does. Here's a fix -- this makes the test not depend on what specific value `exampleData` supplies there, by making the relationship explicit in the code:
```suggestion
    const account1 = eg.makeAccount({ userId: eg.action.realm_init.data.user_id });
```

If we adjusted `exampleData` so it used `selfUser.user_id` there, then an alternative would be:
```js
    const account1 = eg.selfAccount;
```
But the explicit version works in any case.